runtimes/alibi-detect/mlserver_alibi_detect/runtime.py

```diff
@@ -0,0 +1,102 @@
+from mlserver import types
+from mlserver.settings import ModelSettings
+from mlserver.model import MLModel
+from mlserver.errors import InferenceError
+from mlserver.codecs import NumpyCodec
+from fastapi import Request, Response
+from mlserver.handlers import custom_handler
+from .protocols.util import (
+    NumpyEncoder,
+    get_request_handler,
+    Protocol,
+)
+import numpy as np
+import json
+from typing import Optional, Any
+
+# from pydantic import BaseSettings
+
+# ENV_PREFIX_ALIBI_DETECT_SETTINGS = "MLSERVER_MODEL_ALIBI_DETECT_"
+
+
+class AlibiDetectSettings(dict):
+    """
+    Parameters that apply only to alibi detect models
+    """
+
+    def __init__(self, dictionary):
+        for k, v in dictionary.items():
+            setattr(self, k, v)
+
+    # class Config:
+    #     env_prefix = ENV_PREFIX_ALIBI_DETECT_SETTINGS
+
+    init_detector: bool = False
+    detector_type: str = ""
+    protocol: Optional[str] = "seldon.http"
+    init_parameters: Optional[dict] = {}
+    predict_parameters: Optional[dict] = {}
+
+
+class AlibiDetectRuntime(MLModel):
+    """
+    Implementation of the MLModel interface to load and serve `alibi-detect` models.
+    """
+
+    def __init__(self, settings: ModelSettings):
+
+        self.alibi_detect_settings = AlibiDetectSettings(settings.parameters.extra)
+        super().__init__(settings)
+
+    @custom_handler(rest_path="/")
+    async def detect(self, request: Request) -> Response:
+        """
+        This custom handler is meant to mimic the behaviour prediction in alibi-detect
+        """
+        raw_data = await request.body()
+        as_str = raw_data.decode("utf-8")
+
+        try:
+            body = json.loads(as_str)
+        except json.decoder.JSONDecodeError as e:
+            raise InferenceError("Unrecognized request format: %s" % e)
+
+        request_handler = get_request_handler(
+            Protocol(self.alibi_detect_settings.protocol), body
+        )
+        request_handler.validate()
+        input_data = request_handler.extract_request()
+
+        y = await self.predict_fn(input_data, {})
+        output_data = json.dumps(y, cls=NumpyEncoder)
```

Comment:
Instead of the default `json` encoder here, we could leverage `orjson` which should be slightly faster. The interface should be similar to the `json` module, so hopefully it should be a drop-in replacement.